File: backend/examples/report_error.py
import requests

API_TOKEN = "your-project-api-token"
BASE_URL = "http://localhost:5000/api/v1"


def report_error(exception_type, message, stack_trace=None,
                 severity="error", environment="unknown", context=None):
    resp = requests.post(
        f"{BASE_URL}/errors",
        headers={"X-API-Token": API_TOKEN},
        json={
            "exception_type": exception_type,
            "message": message,
            "stack_trace": stack_trace,
            "severity": severity,
            "environment": environment,
            "context": context,
        },
    )
    return resp.json()


if __name__ == "__main__":
    result = report_error(
        exception_type="ValueError",
        message="invalid input data",
        stack_trace=(
            "Traceback (most recent call last):\n"
            '  File "/app/services/user.py", line 42, in create_user\n'
            "    validate_email(email)\n"
            '  File "/app/utils/validators.py", line 15, in validate_email\n'
            '    raise ValueError("invalid input data")'
        ),
        severity="error",
        environment="production",
        context={
            "user_id": "12345",
            "request_path": "/api/users",
            "method": "POST",
        },
    )
    print(result)

File: backend/examples/report_error.js
const API_TOKEN = "your-project-api-token";
const BASE_URL = "http://localhost:5000/api/v1";

async function reportError(exceptionType, message, options = {}) {
  const resp = await fetch(`${BASE_URL}/errors`, {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      "X-API-Token": API_TOKEN,
    },
    body: JSON.stringify({
      exception_type: exceptionType,
      message: message,
      stack_trace: options.stackTrace || undefined,
      severity: options.severity || "error",
      environment: options.environment || "unknown",
      context: options.context || undefined,
    }),
  });
  return resp.json();
}

// 最简调用
const result1 = await reportError("ValueError", "invalid input data");
console.log(result1);

// 完整调用
const result2 = await reportError(
  "ConnectionError",
  "database connection timeout after 30s",
  {
    stackTrace:
      "Traceback (most recent call last):\n" +
      '  File "/app/db/pool.py", line 22, in get_connection\n' +
      "    conn = pool.acquire(timeout=30)\n" +
      '  File "/app/db/pool.py", line 45, in acquire\n' +
      '    raise ConnectionError("database connection timeout after 30s")',
    severity: "critical",
    environment: "production",
    context: { db_host: "pg-master.internal", db_port: 5432, pool_size: 10 },
  }
);
console.log(result2);
